integration-cli/docker_cli_search_test.go

```diff
@@ -10,90 +11,45 @@ import (
 func (s *DockerSuite) TestSearchOnCentralRegistry(c *check.C) {
 	testRequires(c, Network, DaemonIsLinux)
 
-	out, exitCode := dockerCmd(c, "search", "busybox")
-	if exitCode != 0 {
-		c.Fatalf("failed to search on the central registry: %s", out)
-	}
-
-	if !strings.Contains(out, "Busybox base image.") {
-		c.Fatal("couldn't find any repository named (or containing) 'Busybox base image.'")
-	}
+	out, _ := dockerCmd(c, "search", "busybox")
+	c.Assert(out, checker.Contains, "Busybox base image.", check.Commentf("couldn't find any repository named (or containing) 'Busybox base image.'"))
 }
 
 func (s *DockerSuite) TestSearchStarsOptionWithWrongParameter(c *check.C) {
-	out, exitCode, err := dockerCmdWithError("search", "--stars=a", "busybox")
-	if err == nil || exitCode == 0 {
-		c.Fatalf("Should not get right information: %s, %v", out, err)
-	}
-
-	if !strings.Contains(out, "invalid value") {
-		c.Fatal("couldn't find the invalid value warning")
-	}
+	out, _, err := dockerCmdWithError("search", "--stars=a", "busybox")
+	c.Assert(err, check.NotNil)
+	c.Assert(out, checker.Contains, "invalid value", check.Commentf("couldn't find the invalid value warning"))
 
-	out, exitCode, err = dockerCmdWithError("search", "-s=-1", "busybox")
-	if err == nil || exitCode == 0 {
-		c.Fatalf("Should not get right information: %s, %v", out, err)
-	}
-
-	if !strings.Contains(out, "invalid value") {
-		c.Fatal("couldn't find the invalid value warning")
-	}
+	out, _, err = dockerCmdWithError("search", "-s=-1", "busybox")
+	c.Assert(err, check.NotNil)
+	c.Assert(out, checker.Contains, "invalid value", check.Commentf("couldn't find the invalid value warning"))
 }
 
 func (s *DockerSuite) TestSearchCmdOptions(c *check.C) {
 	testRequires(c, Network)
 
-	out, exitCode := dockerCmd(c, "search", "--help")
-	if exitCode != 0 {
-		c.Fatalf("failed to get search help information: %s", out)
-	}
-
-	if !strings.Contains(out, "Usage:\tdocker search [OPTIONS] TERM") {
-		c.Fatalf("failed to show docker search usage: %s", out)
-	}
-
-	outSearchCmd, exitCode := dockerCmd(c, "search", "busybox")
-	if exitCode != 0 {
-		c.Fatalf("failed to search on the central registry: %s", outSearchCmd)
-	}
+	out, _ := dockerCmd(c, "search", "--help")
+	c.Assert(out, checker.Contains, "Usage:\tdocker search [OPTIONS] TERM", check.Commentf("failed to show docker search usage: %s", out))
```

Comment:
`out` is a duplicate here in the comment since the error output will already display `out` as part of the checker output.